server/buildbuddy_server/buildbuddy_server.go

```diff
@@ -191,8 +192,24 @@ func assembleURL(host, scheme, port string) string {
 }
 
 func (s *BuildBuddyServer) getGroupAPIKey(ctx context.Context) string {
+	groupID := ""
+	reqCtx := requestcontext.ProtoRequestContextFromContext(ctx)
```

Comment:
Yep I'm aware of that pattern, still training myself to overcome the awkward syntax and recognize when the idiom can be applied, since it does have the benefit you mentioned. Done!